voltdbclient/conn.go

```diff
@@ -32,72 +32,49 @@
 package voltdbclient
 
 import (
-	"bytes"
 	"database/sql/driver"
 	"errors"
 	"fmt"
-	"io"
+	"math/rand"
 	"net"
-	"sync"
-	"sync/atomic"
 	"time"
 )
 
-// qHandle is a var
-var qHandle int64 = 0 // each query has a unique handle.
-
-// connectionData are the values returned by a successful login.
-type connectionData struct {
-	hostId      int32
-	connId      int64
-	leaderAddr  int32
-	buildString string
-}
-
-type connectionState struct {
-	connInfo      string
-	reader        io.Reader
-	writer        io.Writer
-	connData      connectionData
-	asyncsChannel chan voltResponse
-	asyncs        map[int64]*voltAsyncResponse
-	asyncsMutex   sync.Mutex
-	nl            *networkListener
-	nlwg          *sync.WaitGroup
-	isOpen        bool
-}
-
-// VoltConn represents a connection to VoltDB that can be used to execute
-// queries and other statements.  A VoltConn is initially created with a call
-// to OpenConn.
-//
-// A VoltConn is a driver.Conn; VoltConn also supports an asynchronous api.
-//
-// The creation of a VoltConn represents the creation of two associated
-// goroutines.  One of these is used to listen for responses from the VoltDB
-// server.  The other is used to process responses to asynchronous requests.
-//
-// A VoltConn should not be shared among goroutines; this is true of
-// driver.Conn as well.  But a client can create many instances of a VoltConn.
 type VoltConn struct {
-	cs *connectionState
+	ncs  []*nodeConn
+	r    *rand.Rand
+	open *bool
 }
 
```

Comment:
I am thinking instead of provide a rand for doing scheduling here, we can give a distributor interface. (kind like java client distributor)

This distributor is response for routing work for specific server connection. We can implement query mapping to server logic (client affinity), ratelimit and backpressure in this distributor.